lib/createGltf.js

```diff
@@ -372,40 +356,41 @@ function createMetallicRoughnessTexture(gltf, metallicImage, roughnessImage, occ
     var height = dimensions[1];
     var pixelsLength = width * height;
     var pixels = Buffer.alloc(pixelsLength * 4, 0xFF); // Initialize with 4 channels, unused channels will be white
+    var scratchChannel = Buffer.alloc(pixelsLength);
 
     if (packMetallic) {
         // Write into the B channel
-        var metallicChannel = getImageChannel(metallicImage, 0, width, height);
-        writeChannel(pixels, metallicChannel, 2, width, height);
+        var metallicChannel = getImageChannel(metallicImage, 0, width, height, scratchChannel);
```

Comment:
`scratchChannel` will stay the same size, but sometimes `scratchResizeChannel` may need to be re-allocated. I think this is okay given that best practice is to pass textures that are the same dimensions.